app/tests/src/wallet/account-manager.js

```diff
@@ -45,62 +40,40 @@ const initAccountManager = async (settings, i) => {
     config,
   })
 
-  accountManagers.push(
-    AccountManager({config, cryptoProvider, blockchainExplorer, maxAccountIndex})
-  )
+  return AccountManager({config, cryptoProvider, blockchainExplorer, maxAccountIndex})
 }
 
-before(async () => {
-  for (const accountManagerSetting of accountManagerSettings) {
-    await initAccountManager(accountManagerSetting)
-  }
+before(() => {
+  Object.entries(accountManagerSettings).forEach(([name, setting]) => {
+    accountManagers[name] = initAccountManager(setting)
+  })
 })
 
-describe('account discovery', () => {
-  // TODO: refactor to foreach
-  it('should discover the right amount of accounts', async () => {
-    const mockNet = mockNetwork(ADALITE_CONFIG)
-    mockNet.mockBulkAddressSummaryEndpoint()
-    const accounts = await accountManagers[0].discoverAccounts()
-    // console.log(accounts, accountManagerSettings[0])
-    assert.equal(accounts.length, accountManagerSettings[0].numberOfDiscoveredAccounts)
-    mockNet.clean()
-  })
-  it('should not discover accounts if bulk export is disabled', async () => {
-    const mockNet = mockNetwork(ADALITE_CONFIG)
-    mockNet.mockBulkAddressSummaryEndpoint()
-    const accounts = await accountManagers[1].discoverAccounts()
-    // console.log(accounts, accountManagerSettings[1])
-    assert.equal(accounts.length, accountManagerSettings[1].numberOfDiscoveredAccounts)
-    mockNet.clean()
-  })
-  it('should not discover accounts if wallet is not shelley compatible', async () => {
-    const mockNet = mockNetwork(ADALITE_CONFIG)
-    mockNet.mockBulkAddressSummaryEndpoint()
-    const accounts = await accountManagers[2].discoverAccounts()
-    assert.equal(accounts.length, accountManagerSettings[2].numberOfDiscoveredAccounts)
-    mockNet.clean()
-  })
-  it('should discover one account if no accounts were used', async () => {
-    const mockNet = mockNetwork(ADALITE_CONFIG)
-    mockNet.mockBulkAddressSummaryEndpoint()
-    const accounts = await accountManagers[3].discoverAccounts()
-    assert.equal(accounts.length, accountManagerSettings[3].numberOfDiscoveredAccounts)
-    mockNet.clean()
-  })
+describe('Account discovery', () => {
+  Object.entries(accountManagerSettings).forEach(([name, setting]) =>
+    it(`should discover the right amount of accounts ${name}`, async () => {
+      const accountManager = await accountManagers[name]
+      const mockNet = mockNetwork(ADALITE_CONFIG)
+      mockNet.mockBulkAddressSummaryEndpoint()
+      const accounts = await accountManager.discoverAccounts()
+      assert.equal(accounts.length, setting.numberOfDiscoveredAccounts)
```

Comment:
settings.expectedNumberOfDiscoveredAccounts would be a more suitable to name to make it clear at first sight what's the returned value and what's the expected one

Reply:
Will do :+1: